### DISCUSSION TITLE
Is it risky to use jQuery from a CDN in my script?

### DISCUSSION THREAD

**Discussion by OutshineIssue** [violentmonkey/violentmonkey#2241] [General]
jQuery has had security vulnerabilities in the past, and using a CDN like https://code.jquery.com/jquery-3.7.1.min.js adds extra risks since the domain could be compromised too. Unfortunately, I need it to get my latest script to work, so I have to ask: How risky is it for me to use jQuery from a CDN, considering these scenarios?

**Comment by tophf** [comment]
Until we implement #1558 the only 100% reliable method is to embed the code in the userscript. You may want to use a smaller library though, not the vanilla jQuery.
